### PR TITLE
Multiple flutters sample android

### PR DESCRIPTION
**Pull request by gaaclarke**
issue: https://github.com/flutter/flutter/issues/74530

### CODE REVIEW

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt:12]: don't know if we talked offline before or after this PR
 
```kotlin
lateinit val engines: FlutterEngineGroup

override fun onCreate() {
  super.onCreate()
  engines = ...
}
```

might solve the issue you had

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:22]: Kotlin ftw!!

Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:22]: Same comment as swift. Should this be a set? Since on Android it's not happening on an SDK class where the lifecycle is very clear, alternatively add some state assertions to the consuming class?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:25]: oh man, kotlin ftw!

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:28]: you want to do the same thing as iOS here? (e.g. cull the dead references)?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:17]: you can use named arguments to disambiguate

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:19]: I think the lateinit val pattern might be a bit clearer (since it's less ambiguous by what time they become not null). Lazy is better for things we might indeterministically (or never) access, which isn't the case here.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:38]: I don't think you need to specify the type twice

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:28]: though for fwiw, if you use the engine cache, then things will never be gc'ed under the weak reference. I'd probably just stick to explicit destructions which make things more manageable in the JVM anyway.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:62]: Could we do this for iOS too for consistency?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:69]: bottom

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:70]: maybe this is where we remove the engine cache?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:33]: these are all val right?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:38]: ditto, you may not need to type it twice

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:40]: might not need this

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:43]: ah that's where engines is lazy built first. I think this is an example of the init lifecycle not leaking outside the class and being more deterministic using lateinit vs lazy

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:53]: somewhat surprised this didn't ask you to null check it if it's a var

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:79]: doc

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:19]: private lateinit val

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:18]: I like the swift name better. It's not so much of a "main" activity since it'll cycle back over and over again.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:20]: what's ident?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:22]: add a comment?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:44]: comment

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:43]: add some docs to describe where this would visually come from

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt:26]: ultra-nit: indent

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt:15]: lateinit

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/res/layout/activity_main.xml
Comment (xster) [inline add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/res/layout/activity_main.xml:15]: does the ui say 

count:
hello world!

initially?

**Review by xster — COMMENTED** [review]
nice! kotlin looks pretty clean!

File: add_to_app/multiple_flutters/multiple_flutters_android/app/build.gradle
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/build.gradle:8]: No idea.  Android studio added it, maybe when I was trying to solve my signing issue for release?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt:12]: That would work, too.  I'm not sure if it is preferable to the lazy evaluation.  I googled it a bit and it isn't clear.  I can see some arguments either way.  I believe it's required just because you can't call abstract methods in the constructor, so literally any time after the constructor should be fine.  If that was not the case I think lateinit would be ideal since it would crash a bit earlier in a situation where we can't completely protect ourselves from crashing.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:22]: Oh man, so much easier in this case.  Swift's protocols are weird.

Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:22]: It isn't necessary for this example to function correctly.  I think a designer of a system for production code would have to make the semantics very clear about what happens when you double subscribe the same object.  Another option is just to throw an exception if someone double subscribes.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt:28]: I'm doing the same thing I'm doing on iOS.  Skip on notify, cull on removeObserver.

This should handle anybody that is being GC'd or manually deleted.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:38]: removed

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:62]: It already is happening, but in a slightly different way.  On iOS SingleFlutterViewController bundles the bindings with itself so it's not necessary.  I wouldn't want to add a concept that isn't necessary.  I don't think that would make the sample more clear.

A more similar approach would be to have Fragment that bundles the communication with a FlutterFragment, then have SigqleFlutterActivity and DoubleFlutterActivity embed that Fragment.  Unfortunately for me, converting an Activity to a Fragment isn't trivial.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:19]: There is no such thing as `latinit val`, there is only `lateinit var`, that's why I think lazy is better here.  Kotlin doesn't have the idea of `final` either.  From a user's perspective they should never need to think about the value ever being null.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:69]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:70]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:33]: yep, done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:53]: Hmm, me too.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:79]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt]: Renamed, mainActivityIdentifier

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:20]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:42]: Ok, I switched engines to lateinit.  If this was code that I was writing for me I would prefer lazy evals, but having a lazy eval for a variable is a bit much for sample code.

I kept lazy for the engine bindings because they are actually used before onCreate in the case of SingleFlutterActivity, so the code is cleaner this way.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt]: done

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt]: done, reformatted all the code automatically.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt]: mentioned above my these should still be lazy

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/res/layout/activity_main.xml
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/res/layout/activity_main.xml:15]: Nope

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt
Comment (gaaclarke) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/App.kt]: Switch it, explained below.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/build.gradle
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/build.gradle:8]: I think we can remove it

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt]: Either way is ok. I'd just add some comments or docs saying we're taking precautions with the engine bindings or this is a simple demo and you might run into trouble in these circumstances.

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DoubleFlutterActivity.kt:62]: Ya, I just meant generally where the platform channel handlers sit between the 2 platforms if someone was looking for them in each of the platforms. If they need to be in different sounding files, maybe add a comment cross-referencing the difference?

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/EngineBindings.kt:45]: SG

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/MainActivity.kt:15]: There's still this bit (maybe you didn't want to do a rename in the same commit)

File: add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt
Comment (xster) [reply add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/SingleFlutterActivity.kt]: ack

**Comment by xster** [comment]
Didn't go through the complete code again yet. I think my only remaining comment was the MainActivity name. Otherwise I think this is ready to move out of draft after your folder move.

**Comment by gaaclarke** [comment]
@xster Do you know how the CI works?  Looks like the version of Flutter that the tests are being run against don't have access to the new API.  It's choking on "FlutterEngineGroup".

**Comment by xster** [comment]
Oh wait, right. @RedBrogdon I'm somewhat surprised the iOS PR passed the CI. 
The idea was that the samples must be buildable using the latest stable branch unless the samples are in the experimental folder. But I'm not sure why that test didn't fail with the iOS PR (which is using FlutterEngineGroup, not in 1.22). 

As a solution, I think it'd be ok if we waited until next month to click the merge button once the stable is pushed. @RedBrogdon I don't know if there's otherwise a cheaper solution than moving everything to experimental then moving back. i.e. some flag we can maybe set in the test file?

**Comment by RedBrogdon** [comment]
> Looks like the intent is that things are in experimental if they use API's that haven't landed on stable.

That's correct. One of the invariants, so to speak, of the repo is that every root-level project should build/test with the stable channel, and everything that can't should be in `/experimental`.

> Oh wait, right. @RedBrogdon I'm somewhat surprised the iOS PR passed the CI.

Y'all just uncovered a bug in our CI setup. Looks like `xcpretty` is swallowing error codes returned by `xcodebuild`. We should be configuring it to return the error code instead:

https://www.rubydoc.info/gems/xcpretty/0.2.8#usage

I'll create a separate PR to fix it.

> As a solution, I think it'd be ok if we waited until next month to click the merge button once the stable is pushed. @RedBrogdon I don't know if there's otherwise a cheaper solution than moving everything to experimental then moving back. i.e. some flag we can maybe set in the test file?

You can certainly leave a PR open, since it's unlikely to run into merge conflicts. That said, the iOS version of the app has landed, and I can't fix the hole you just identified in our CI without dealing with that fact. 😄 

That leaves a few options:

* Move `multiple_flutters` to experimental and land the Android version there as well.
* Remove `multiple_flutters` entirely, add the iOS and Flutter versions to this PR (unwieldy, but doable), and leave this PR open for now.
* Leave `multiple_flutters` in place, remove it from the Flutter and iOS CI project lists, and leave this PR open for now.
* Remove `multiple_flutters`, add it to this PR, then land this PR in the `beta` branch of this repo, which is where we're getting samples for the next release ready.

WDYT? I hadn't realized y'all were targeting anything other than stable, or I would have mentioned this earlier.

**Comment by gaaclarke** [comment]
I don't have a good idea about when the API is hitting stable, but a 5th option is that we pause the train until the API hits stable.  That means this doesn't land and your iOS change doesn't land.  That's the least amount of work for everyone but might not be ideal depending on the timeline for the next stable release.  Looks like the last one was 1/25/21 so it should't be long now I imagine.

**Comment by xster** [comment]
Waiting to merge sounds ok to me. That would be the simplest thing to do.

**Comment by domesticmouse** [comment]
I'd suggest pull from master to fix the Master Branch CI stable channel failures

**Review by RedBrogdon — APPROVED** [review]
Looks like Flutter tests are failing because they're pulling in older master code that's not touched by this PR. I'm going to call that a quirk of how GHA works and not a problem with this code, and then do the merge anyway.

**Comment by saurabhchalke** [comment]
Is there a way to start with a Flutter screen at the beginning and bypassing any Kotlin screens at all?
I am trying to tweak this example with Flutter -> Native -> Flutter flow.

Also, the Kotlin to Flutter navigation seems to be slow on my Phone (Mi A2, Android 10), although Flutter to Kotlin seems to be quick.
Do you see any optimizations possible in here?

**Comment by gaaclarke** [comment]
> I Is there a way to start with a Flutter screen at the beginning and bypassing any Kotlin screens at all?

Yep, I don't see any reason why it shouldn't work.

> Also, the Kotlin to Flutter navigation seems to be slow on my device (Mi A2, Android 10), although Flutter to Kotlin seems to be quick.
Do you see any optimizations possible in here?

It is slow on debug builds and on emulators, on release builds on real devices it is fast.  If you found a scenario where the transition isn't fast on release builds / real devices, please file an issue with the supporting evidence and reproduction steps.

**Pull request merged**
